### PR TITLE
Define the scope of a display control with number of encounters

### PR DESCRIPTION
**Pull request by shivarach**
https://msfprojects.atlassian.net/browse/MOBN-1211

### CODE REVIEW

File: bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/service/BahmniObsService.java [hunk 1]
```diff
@@ -11,25 +11,26 @@
...
-    public Collection<BahmniObservation> observationsFor(String patientUuid, Concept rootConcept, Concept childConcept, Integer numberOfVisits, Date startDate, Date endDate, String patientProgramUuid);
-
-    public Collection<BahmniObservation> getObservationForVisit(String visitUuid, List<String> conceptNames, Collection<Concept> obsIgnoreList, Boolean filterOutOrders, Order order);
-    public Collection<BahmniObservation> getObservationsForEncounter(String encounterUuid, List<String> conceptNames);
-    public Collection<BahmniObservation> getObservationsForPatientProgram(String patientProgramUuid, List<String> conceptNames, List<String> obsIgnoreList);
-    public Collection<BahmniObservation> getLatestObservationsForPatientProgram(String patientProgramUuid, List<String> conceptNames, List<String> obsIgnoreList);
-    public Collection<BahmniObservation> getInitialObservationsForPatientProgram(String patientProgramUuid, List<String> conceptNames, List<String> obsIgnoreList);
```
Comment (shivarach) [inline bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/service/BahmniObsService.java:32]: just public keyword is removed which is optional in interface

File: bahmnicore-api/src/test/java/org/bahmni/module/bahmnicore/dao/impl/VisitDaoImplIT.java [hunk 2]
```diff
@@ -50,10 +50,47 @@ public void shouldNotGetVoidedEncounter() throws Exception {
     }
 
     @Test
-    public void shouldReturnAllEncountersWhenNumberOfVisitsAndEncountersAreNotProvided() {
+    public void shouldReturnAllNonVoidedEncountersWhenNumberOfVisitsAndEncountersAreNotProvided() {
```
Comment (binduak) [inline bahmnicore-api/src/test/java/org/bahmni/module/bahmnicore/dao/impl/VisitDaoImplIT.java:53]: @shivarach what do you mean by NonVoidedEncounters ??

File: bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/dao/impl/ObsDaoImpl.java [hunk 3]
```diff
@@ -106,17 +119,19 @@
             query.append(" order by obs.obsDatetime desc ");
         }
 
-
         Query queryToGetObservations = sessionFactory.getCurrentSession().createQuery(query.toString());
-        queryToGetObservations.setMaxResults(limit);
+        queryToGetObservations.setMaxResults(limit != null ? limit : -1);
```
Comment (binduak) [inline bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/dao/impl/ObsDaoImpl.java:123]: what does -1 limit value means here ??

File: bahmnicore-api/src/test/java/org/bahmni/module/bahmnicore/dao/impl/VisitDaoImplIT.java (hunk 2 above)
Comment (shivarach) [reply bahmnicore-api/src/test/java/org/bahmni/module/bahmnicore/dao/impl/VisitDaoImplIT.java:53]: Encounters that are not voided. The query specifically looks for non-voided encounters.

File: bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/dao/impl/ObsDaoImpl.java (hunk 3 above)
Comment (shivarach) [reply bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/dao/impl/ObsDaoImpl.java:123]: Setting a negative number to hibernate `Query` will return all the reocrds

**Review by binduak — APPROVED** [review]

**Pull request merged**